test/pool.test.ts

```diff
@@ -738,6 +738,30 @@ describe("ReceiverWeights", function () {
     assert(weights.length == 0);
   });
 
+  it("Allows adding items after removing all items", async function () {
+    // Add an item and then clear the list
+    const weights_test = await deployReceiverWeightsTest();
+    const [addr1] = randomAddresses();
+
+    await weights_test.setWeights([
+      {receiver: addr1, weight: 1},
+      {receiver: addr1, weight: 0},
+    ]);
+
+    assert((await weights_test.receiverWeightsSumDelta()).eq(1 - 1));
+    let weights = await weights_test.getReceiverWeightsIterated();
+    assert(weights.length == 0);
+
+    // Add an item
+    await weights_test.setWeights([{receiver: addr1, weight: 2}]);
+
+    assert((await weights_test.receiverWeightsSumDelta()).eq(2));
+    weights = await weights_test.getReceiverWeightsIterated();
+    assert(weights.length == 1);
+    assert(weights[0].receiver == addr1);
+    assert(weights[0].weight == 2);
```

Comment:
```suggestion
    assert.deepEqual(weights, { receiver: addr1, weight: 2 });
```

Reply:
I couldn't use it in code, could you help me out with this?

`assert.deepEqual(weights, [{receiver: addr1, weight: 2}]);`
causes a compilation error:
`Type '{ receiver: string; weight: number; }' is missing the following properties from type '{ receiver: string; weight: number; 0: string; 1: number; }': 0, 1`

So I guess that the silly fix for that should be:
`assert.deepEqual(weights, [{receiver: addr1, weight: 2, 0: addr1, 1: 2}]);`
it compiles, but now the test fails:
```
      AssertionError: expected [ Array(1) ] to deeply equal [ Array(1) ]
      + expected - actual

       [
      -  [
      -    "0x1B78D18678cfA961f98E7DC824e9ABD046b0c594"
      -    2
      -  ]
      +  {
      +    "0": "0x1B78D18678cfA961f98E7DC824e9ABD046b0c594"
      +    "1": 2
      +    "receiver": "0x1B78D18678cfA961f98E7DC824e9ABD046b0c594"
      +    "weight": 2
      +  }
       ]
```

Ok, an array then:
`assert.deepEqual(weights, [[addr1, 2]]);`
but it again fails to compile with:
`Type '[string, number]' is not assignable to type '{ receiver: string; weight: number; 0: string; 1: number; }'.`

Reply:
I see. I assumed etherjs just returned an object. But if it returns this more complicated struct I’m ok if we leave it like it is.